src/native/fs/fs_napi.cpp

```diff
@@ -30,6 +30,21 @@
     #include <sys/statfs.h>
 #endif
 
+#define ROUNDUP(X,Y) ( (Y) * ( ((X)+(Y)-1) / (Y) ) )
+
+#define GPFS_DOT_ENCRYPTION_EA "Encryption"
+#define GPFS_ATTR_INTERNAL_APPL_ID   3
+#define GPFS_ATTR_EXT_NAMESPACE_ID   4
+
+#define GPFS_XATTR_PREFIX "gpfs"
+#define XATTR_APPID_TO_NAMESPACE(id) \
+  (id == GPFS_ATTR_INTERNAL_APPL_ID) ? GPFS_XATTR_PREFIX : ""
+#define XATTR_APPID_TO_NAMESPACE_LEN(id) \
+  (sizeof(XATTR_APPID_TO_NAMESPACE(id)) - 1)
+#define GPFS_XATTR_FULL_NAME GPFS_XATTR_PREFIX "." GPFS_DOT_ENCRYPTION_EA
+// Should total to 256 (sizeof(buffer) 212 + sizeof(header) 16 + sizeof(payload) 28)
+#define GPFS_XATTR_BUFFER_SIZE   212 
```

Comment:
I would make this an assert in fs_napi() that `sizeof(gpfsRequest_t)` correctly sums up to 256